### ISSUE TITLE
Intermittant Crashes during World Generation (Botania)

### ISSUE THREAD

**Issue by 0xKate** [embeddedt/ArchaicFix#10]
Hi, I have been having a rare intermittent crash on world generation.

So far this seems to only happen while I'm testing in singleplayer. I have yet to have this happen on a dedicated server.

Mods seemingly involved (top down):
Botania
ArchaicFix
tclproject.immersivecavegen
tclproject.mysteriumlib

I was testing generation in Aroma1997sDimension the last time it crashed (although it crashes on the overworld sometimes too). I believe this is the Botania Mushrooms that are in question as they are the only flower that generates in caves and has a light level.

Let me know if there is anything I should try to disable for compatibility I couldn't find anything in the configs about lighting tweaks.

[crash-2022-08-22_18.08.37-server.txt](https://github.com/embeddedt/ArchaicFix/files/9411734/crash-2022-08-22_18.08.37-server.txt)
[crash-2022-08-23_18.58.38-server.txt](https://github.com/embeddedt/ArchaicFix/files/9411735/crash-2022-08-23_18.58.38-server.txt)
[fml-client-latest.log](https://github.com/embeddedt/ArchaicFix/files/9411758/fml-client-latest.log)

Thanks for any help!

**Comment by embeddedt** [comment]
Please update to the latest development version of ArchaicFix (link in the README) and try this again.

**Comment by 0xKate** [comment]
Okay will do.

**Comment by 0xKate** [comment]
I am unsure If another core mod is breaking things or what's goin on, but the latest version from the nightly link in the readme is crashing on startup.

archaicfix-mc1.7.10-0.2.0-8-g5d94501.jar
SHA256: e6c8615afe68e2492d8909160b3213cd3f0c7501ab240960911210c6818b0bac

[fml-client-latest.log](https://github.com/embeddedt/ArchaicFix/files/9411901/fml-client-latest.log)

Thanks for your assistance!

**Comment by 0xKate** [comment]
Unfortunately, updating ArchaicFix requires updating FalsePattern Lib, doing so leads to a rabbit hole of crashes.

1. Breaks Triangulator <-> Neodymium compat: (even if updating to latest)
[fml-client-1.log](https://github.com/embeddedt/ArchaicFix/files/9412163/fml-client-1.log)

2. If I disable Triangulator completely: Archaic Fix crashes from CraftingManager:
[crash-2022-08-23_21.13.00-client.txt](https://github.com/embeddedt/ArchaicFix/files/9412164/crash-2022-08-23_21.13.00-client.txt)

2.b Further Investigation Shows this update is incompatible with CoFH Core:
[fml-client-latest.log](https://github.com/embeddedt/ArchaicFix/files/9412162/fml-client-latest.log)

Unfortunately everything works great with the older versions other than this Botania crash (which can be worked around by disabling mushroom generation) Updating FalsePatternLib starts to break many things.

**Comment by embeddedt** [comment]
Don't update FalsePatternLib yet. As you've seen, it triggers other problems, and we're still working on getting that to work.

Independently of that, I updated the mod's build script recently, and apparently that is already enough to start causing compatibility issues. :confused: 

Try this development build, I've reverted the build script update till I figure out the problem.

https://github.com/embeddedt/ArchaicFix/suites/7960616441/artifacts/340171113
